### DISCUSSION TITLE
Unmarshaller on []byte field

### DISCUSSION THREAD

**Discussion by yinebebt** [spf13/viper#1907] [Q&A]
Does Viper support unmarshalling a Base64 string value into []byte? I mean like the `json/encoding` unmarshaller do. 
For example, if I have a struct: `type config struct { Salt []byte }` can Viper unmarshal a configuration like `salt: <base64-encoded-string>` directly into the Salt field?

**Comment by sagikazarmark** [comment]
You could implement that by introducing a custom type using [mapstructure](https://github.com/go-viper/mapstructure/).

You can find some examples here: https://sagikazarmark.hu/blog/decoding-custom-formats-with-viper/
